examples/ecdsa/circuit.rs

```diff
@@ -0,0 +1,286 @@
+use std::marker::PhantomData;
+
+use bellperson::{
+  gadgets::{boolean::AllocatedBit, num::AllocatedNum},
+  ConstraintSystem, SynthesisError,
+};
+use ff::{PrimeField, PrimeFieldBits};
+use generic_array::typenum::U8;
+use neptune::{
+  circuit::poseidon_hash,
+  poseidon::{Poseidon, PoseidonConstants},
+};
+use nova_snark::{
+  gadgets::{ecc::AllocatedPoint, utils::alloc_num_equals},
+  traits::StepCircuit,
+};
+use subtle::Choice;
+
+// An ECDSA signature proof that we will use on the primary curve
+#[derive(Clone, Debug)]
+pub struct EcdsaCircuit<F>
+where
+  F: PrimeField<Repr = [u8; 32]>,
+{
+  pub z_rx: F,
```

Comment:
@arthurgreef do we need a separate field element that tracks `is_infinity` when representing ECC points in affine form?